GT-96 Django monitoring app

Letizia97 opened the pull request:
Features:
- improvements for Python 3 compatibility
- sync table
- sync table with submithosts

Letizia97 committed:
Ensure Python3 compatibility.

Letizia97 committed:
Make imports more explicit.

Letizia97 committed:
Add allowed hosts.

Letizia97 committed:
Ensure Python3 compatibility for CloudSite view.

Letizia97 committed:
Ensure the grid site works when retrieving multiple records.

Letizia97 committed:
Add import for pymysql.

Letizia97 committed:
Add models for the sync page.

Letizia97 committed:
Modify names to ensure consistency with the new models (GridSiteSync).

Letizia97 committed:
Add serializer for gridsync site.

Letizia97 committed:
Modify names to ensure consistency with modified models.

Letizia97 committed:
Add view for the gridsync table, for both single site and all sites.

Letizia97 committed:
Update field names to ensure consistency with modified models and serialisers.

Letizia97 committed:
Add html templates for gridsync, both all sites page and single site page.

Letizia97 committed:
Update field names to ensure consistency with modified views.

Letizia97 committed:
Add urls for gridsync pages.

Letizia97 committed:
Add model for the table with submithosts.

Letizia97 committed:
Add serialiser for the table with submithosts.

Letizia97 committed:
Add view for the sync table showing submithosts.

Letizia97 committed:
Add hyperlink to table showing submithosts.

Letizia97 committed:
Add template for table showing submithosts.

Letizia97 committed:
Add url to the page showing submithosts.

Letizia97 committed:
Add static root to be able to run "manage.py collectstatic".

tofu-rocketry commented:
Can you fix the 'style' category issues that Codeclimate has raised? Don't worry about the other categories of issue for now though, other than the 'clarity' category about the `if` statement.

For line length, follow the [PEP 8](https://peps.python.org/pep-0008/) guidance for longer lines (I need to change the linting to use this setting though):

> Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the line length limit up to 99 characters, provided that comments and docstrings are still wrapped at 72 characters.

Letizia97 committed:
Trim trailing whitespaces.

Letizia97 committed:
Add newline at end of file.

Letizia97 committed:
Remove whitespace(s) where not necessary or too many.

Letizia97 committed:
Add whitespace(s) around operators where necessary.

Letizia97 committed:
Remove spaces around keyword / parameter equals.

Letizia97 committed:
Add space after punctuation where necessary.

Letizia97 committed:
Remove unnecessary blank lines.

Letizia97 committed:
Move import to the top of file.

Letizia97 committed:
Rearrange/refactor imports to shorten lines.

Letizia97 committed:
Rearrange/refactor serialiser fields to shorten lines.

Letizia97 committed:
Rearrange/refactor urlpatterns to shorten lines.

Letizia97 committed:
Rearrange dicts to shorten lines.

Letizia97 committed:
Refactor parts to create/fill dicts, to reduce duplication and file length.

Letizia97 committed:
Bring functions outside class to avoid duplication.

Letizia97 committed:
Refactor to reduce line length.

Letizia97 committed:
Merge if statements to ensure clarity.

Letizia97 committed:
Rearrange SQL queries to reduce line length.

Letizia97 committed:
Create separate function for sync_status to reduce duplication.

Letizia97 committed:
Create separate function foupdating stdout and returncode.

Letizia97 committed:
Refactor to reduce length of lines.

gregcorbett commented on monitoring/publishing/models.py:
```diff
@@ -6,28 +6,60 @@
 class GridSite(models.Model):
     fetched = models.DateTimeField(auto_now=True)
-    name = models.CharField(max_length=255, primary_key=True)
+    SiteName = models.CharField(max_length=255, primary_key=True)
```
Is it a django thing that these kinds of variables need to be CamelCase?

tofu-rocketry replied:
The name of the field needs to match the relevant database, but this only really applies to the unmanaged models that are designed to be used to pull from the APEL Repository dbs. Probably not needed here (though if we revert we'd have to change any other references).

tofu-rocketry replied:
Tracking in #19

gregcorbett commented on monitoring/publishing/serializers.py:
```diff
@@ -11,7 +16,36 @@ class GridSiteSerializer(serializers.HyperlinkedModelSerializer):
+            'SyncStatus'
+        )
+
+        # Sitename substitutes pk
+        lookup_field = 'SiteName'
+        extra_kwargs = {
+            'url': {'lookup_field': 'SiteName'}
+        }
```
what does this do? Are both `lookup_field` and `extra_kwargs` needed?

gregcorbett commented:
Looks good, a few questions - see above. I can see codeclimate raised 47 issues, but can't see what they are 😢

tofu-rocketry committed:
Add missing objects that weren't created/imported

tofu-rocketry committed:
Correct reference to grid database from settings

tofu-rocketry committed:
Automate venv activation

Putting venv activation in the wsgi file means that Apache can invoke
the venv and it doesn't need to be done manually.

tofu-rocketry committed:
Tidy up settings

tofu-rocketry committed:
Change sync status to match old pages

tofu-rocketry committed:
Revert changes to availability app urls

tofu-rocketry committed:
Simplify availability app to just return "OK" 200

tofu-rocketry committed:
Remove year limitation on grid pub SQL query

tofu-rocketry committed:
Add basic HTML template for status page

tofu-rocketry committed:
Add availability app to installed apps

This will hopefully register the templates it contains

tofu-rocketry committed:
Update cutoff date for cloud query
